sycl/doc/EnvironmentVariables.md

```diff
@@ -8,7 +8,7 @@ compiler and runtime.
 | Environment variable | Values | Description |
 | -------------------- | ------ | ----------- |
 | `ONEAPI_DEVICE_SELECTOR` | [See below.](#oneapi_device_selector)  | This device selection environment variable can be used to limit the choice of devices available when the SYCL-using application is run.  Useful for limiting devices to a certain type (like GPUs or accelerators) or backends (like Level Zero or OpenCL).  This device selection mechanism is replacing `SYCL_DEVICE_FILTER` .  The `ONEAPI_DEVICE_SELECTOR` syntax is shared with OpenMP and also allows sub-devices to be chosen. [See below.](#oneapi_device_selector) for a full description.   |
-| `SYCL_DEVICE_ALLOWLIST` | See [below](#sycl_device_allowlist) | Filter out devices that do not match the pattern specified. `BackendName` accepts `host`, `opencl`, `level_zero`, `native_cpu` or `cuda`. `DeviceType` accepts `host`, `cpu`, `gpu` or `acc`. `DeviceVendorId` accepts uint32_t in hex form (`0xXYZW`). `DriverVersion`, `PlatformVersion`, `DeviceName` and `PlatformName` accept regular expression. Special characters, such as parenthesis, must be escaped. DPC++ runtime will select only those devices which satisfy provided values above and regex. More than one device can be specified using the piping symbol "\|".|
+| `SYCL_DEVICE_ALLOWLIST` | See [below](#sycl_device_allowlist) | Filter out devices that do not match the pattern specified. `BackendName` accepts `host`, `opencl`, `level_zero`, `native_cpu` or `cuda`. `DeviceType` accepts `host`, `cpu`, `gpu`, `fpga`, or `acc`. `fpga` and `acc` are handled in the same manner. `DeviceVendorId` accepts uint32_t in hex form (`0xXYZW`). `DriverVersion`, `PlatformVersion`, `DeviceName` and `PlatformName` accept regular expression. Special characters, such as parenthesis, must be escaped. DPC++ runtime will select only those devices which satisfy provided values above and regex. More than one device can be specified using the piping symbol "\|".|
```

Comment:
> `DeviceType` accepts `host`, `cpu`, `gpu`, `fpga`, or `acc`. `fpga` and `acc` are handled in the same manner.

This information is duplicated in this document. I suggest we move it to a separate section or table and reference to it from all the places where `DeviceType` accepted values are listed.

Reply:
@bader Looks like I lost track of this PR.
I see that `DeviceType` is different for ONEAPI_DEVICE_SELECTOR, SYCL_DEVICE_ALLOWLIST, and SYCL_REDUCTION_PREFERRED_WORKGROUP_SIZE.

For ONEAPI_DEVICE_SELECTOR, `DeviceType` is `cpu, gpu, fpga`
For SYCL_DEVICE_ALLOWLIST, `DeviceType` is `cpu, gpu, acc, fpga`
And for SYCL_REDUCTION_PREFERRED_WORKGROUP_SIZE, `DeviceType` is `cpu, gpu, acc`.

Since `DeviceType` is different for different environment variables, I don't see how separating out `DeviceType` to another section and referencing from there would help in reducing redundancy.

Reply:
> Since `DeviceType` is different for different environment variables, 

Why is `DeviecType` different? This sounds like a bug to me. I don't see reasons to have inconsistent values for the same concept for different environments variables.

Reply:
True - it is weird to have different `DeviceType` for different environment variables. One reason for this discrepancy is that, unlike ONEAPI_DEVICE_SELECTOR, SYCL_DEVICE_ALLOWLIST and SYCL_REDUCTION_PREFERRED_WORKGROUP_SIZE are SYCL-specific, and for them we allow using `acc` as a shorthand for `accelerator`, which is one of the device type defined in SYCL spec.
One way to reconcile this is to rename `fpga` to `acc` in ONEAPI_DEVICE_SELECTOR, but, I'm not sure if we should do that now as ONEAPI_DEVICE_SELECTOR will eventually move to UR (AFAIK).

Reply:
I suggest we file a tracker to make the interface consistent for the components we control/inflence.

Reply:
Done: https://github.com/intel/llvm/issues/15793